security/advancedtls/crl_test.go

```diff
@@ -636,17 +641,33 @@ func setupTLSConn(t *testing.T) (net.Listener, *x509.Certificate, *ecdsa.Private
 // TestVerifyConnection will setup a client/server connection and check revocation in the real TLS dialer
 func TestVerifyConnection(t *testing.T) {
 	l, cert, key := setupTLSConn(t)
+
+	done := make(chan interface{}, 1)
+	var wg sync.WaitGroup
+	wg.Add(1)
```

Comment:
Optional: could use a channel here as well, closing it when done.  E.g.

```go
exit := make(chan struct{})
go func() { ...
	switch {
	case <-done:
		close(exit)
		return
```

Reply:
I refactored this to only accept one connection at a time and moved it to the test.Run. Removed all the sync calls